src/erlcloud_redshift.erl

```diff
@@ -0,0 +1,220 @@
+-module(erlcloud_redshift).
+
+-include_lib("erlcloud/include/erlcloud.hrl").
+-include_lib("erlcloud/include/erlcloud_aws.hrl").
+
+-define(API_VERSION, "2012-12-01").
+
+-export([configure/2, configure/3, new/2, new/3]).
+
+-export([describe_clusters/0, describe_clusters/1, describe_clusters/2]).
+
+-spec(new/2 :: (string(), string()) -> aws_config()).
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{access_key_id=AccessKeyID,
+                secret_access_key=SecretAccessKey}.
+
+-spec(new/3 :: (string(), string(), string()) -> aws_config()).
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{access_key_id=AccessKeyID,
+                secret_access_key=SecretAccessKey,
+                route53_host=Host}.
+
+-spec(configure/2 :: (string(), string()) -> ok).
+configure(AccessKeyID, SecretAccessKey) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey)),
+    ok.
+
+-spec(configure/3 :: (string(), string(), string()) -> ok).
+configure(AccessKeyID, SecretAccessKey, Host) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host)),
+    ok.
+
+-spec redshift_query(aws_config(), string(), list({string(), string()})) ->
+    {ok, term()} | {error, term}.
+redshift_query(Config, Action, Params) ->
+    QParams = [{"Action", Action}, {"Version", ?API_VERSION} | Params],
+    erlcloud_aws:aws_request_xml4(get, Config#aws_config.redshift_host,
+                                  "/", QParams, "redshift", Config).
+
+-spec describe_clusters() -> {ok, proplists:proplist()} |
+                             {ok, proplists:proplist(), list()} |
+                             {error, term()}.
+describe_clusters() ->
+    describe_clusters(erlcloud_aws:default_config()).
+
+-spec describe_clusters(list() | aws_config()) -> {ok, proplists:proplist()} |
+                                                  {ok, proplists:proplist(), list()} |
+                                                  {error, term()}.
+describe_clusters(Options) when is_list(Options) ->
+    describe_clusters(Options, erlcloud_aws:default_config());
+describe_clusters(AwsConfig) ->
+    describe_clusters([], AwsConfig).
+
+-spec describe_clusters(list(), aws_config()) -> {ok, proplists:proplist()} |
+                                                 {ok, proplists:proplist(), list()} |
+                                                 {error, term()}.
+describe_clusters(Options, AwsConfig) ->
```

Comment:
please make the marker support correct/better.
http://docs.aws.amazon.com/redshift/latest/APIReference/API_DescribeClusters.html 
there should be following API:
describe_clusters().
describe_clusters(Cfg).
describe_clusters(Ids)
describe_clusters(Ids, Cfg)
describe_clusters(Ids, Options, Cfg).

describe_cluster_all(Options).
describe_cluster_all(Options, Cfg).